crates/pathfinder/src/storage/contract.rs

```diff
@@ -146,7 +144,7 @@ impl ContractsTable {
         hash: ContractHash,
     ) -> anyhow::Result<()> {
         transaction.execute(
-            r"INSERT INTO contracts (address, hash) VALUES (:address, :hash)",
+            r"INSERT OR IGNORE INTO contracts (address, hash) VALUES (:address, :hash)",
```

Comment:
Are multiple deploys of a contract possible or what does lead to this? Aah yes, when reorging ofc... Hmm... But I wonder if the ignore is the right way to go? Technically the address would need to be overwritten with the latest version whenever two deploys have been observed, right? Otherwise it could point to the wrong hash.. But this might open up a can of worms with the contract addresses.

However in this storage model, all global state trees which may have referenced a no longer valid contract hash from previous deploys should now be gone.